shell/platform/darwin/ios/framework/Source/FlutterTextInputPlugin.mm

```diff
@@ -1392,6 +1427,35 @@ - (void)updateEditingState {
   }
 }
 
+- (void)updateEditingStateWithDelta {
+  NSUInteger selectionBase = ((FlutterTextPosition*)_selectedTextRange.start).index;
+  NSUInteger selectionExtent = ((FlutterTextPosition*)_selectedTextRange.end).index;
+
+  // Empty compositing range is represented by the framework's TextRange.empty.
+  NSInteger composingBase = -1;
+  NSInteger composingExtent = -1;
+  if (self.markedTextRange != nil) {
+    composingBase = ((FlutterTextPosition*)self.markedTextRange.start).index;
+    composingExtent = ((FlutterTextPosition*)self.markedTextRange.end).index;
+  }
+
+  NSDictionary* state = @{
+    @"oldText" : [NSString stringWithString:_currentTextEditingDelta.oldText],
+    @"deltaText" : [NSString stringWithString:_currentTextEditingDelta.deltaText],
+    @"deltaStart" : @(_currentTextEditingDelta.deltaStart),
+    @"deltaEnd" : @(_currentTextEditingDelta.deltaEnd),
+    @"selectionBase" : @(selectionBase),
+    @"selectionExtent" : @(selectionExtent),
+    @"selectionAffinity" : @(_selectionAffinity),
+    @"selectionIsDirectional" : @(false),
+    @"composingBase" : @(composingBase),
+    @"composingExtent" : @(composingExtent),
+  };
+
+  [self.textInputDelegate updateEditingClient:_textInputClient withDelta:state];
+  _currentTextEditingDelta = [[FlutterTextEditingDelta alloc] initWithNonText:self.text];
```

Comment:
Is `[[FlutterTextEditingDelta alloc] initWithNonText:self.text]` just to indicate there's no delta? you can set `_currentTextEditingDelta` to nil I think?